### ISSUE TITLE
No status ("complete") when pay is xpay

### ISSUE THREAD

**Issue by hMsats** [ElementsProject/lightning#7923]
v24.11

```
lightning-cli setconfig xpay-handle-pay true
{
   "config": {
      "config": "xpay-handle-pay",
      "value_bool": true,
      "source": "/media/ssd/.lightning_test/bitcoin/config:2",
      "plugin": "/home/user/lightning_test/plugins/cln-xpay",
      "dynamic": true
   }
}

lightning-cli fetchinvoice lno1zcssxxtelekwpe57eekenf9uffanpmfz4r9cwq7a0hfdltmpncr3ylst 103000
{
   "invoice": "lni1qqg8nw78sjant82dfwq92t83qeu3793pqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqk5srqxf9skppq2kgcwnjf3wuwj7dq9kr80mm4azdplsukplv9y7qcvqt8tkgwx36egycqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqkqsr9t9n0h98fc8qwzucvmmqgxug4e8xr3aj2x88y52cak8r8ef4kvqs9tq5ypeys3w9r2q4fuk8shuw8hkeqjxget2vtt72ft3fs927rywuqqez8zxhz6qejeulvq7lm7u7cthndrkgeymhnasunze5pzeaw0khc994sdvn0qrll0r7j7wvxxv3t86ll2x2y8qqqqqqqqqqqqqqqysqqqqqqqqqqqqp6f9jm7k9yqqqqqq2gpr82l0mf2pqkl499zwtaed24kshmqpv4q3m3c24cgwjgujf2c0d9qjh950xf2k25qcpjfvtqggrr9ulum8qu60vumve5j7y57esa5323ju8q0whm5kl4aseupcj0c9lqsxd2npltp2tk26ml2a2dpjhkp6lcllev4ckvfytwhvcsr49xn44zlgl07gzwym3ldccjhlcf69zwmhzafa3avdcmvt3ejkztxg7xcva6",
   "changes": {}
}

lightning-cli pay lni1qqg8nw78sjant82dfwq92t83qeu3793pqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqk5srqxf9skppq2kgcwnjf3wuwj7dq9kr80mm4azdplsukplv9y7qcvqt8tkgwx36egycqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqkqsr9t9n0h98fc8qwzucvmmqgxug4e8xr3aj2x88y52cak8r8ef4kvqs9tq5ypeys3w9r2q4fuk8shuw8hkeqjxget2vtt72ft3fs927rywuqqez8zxhz6qejeulvq7lm7u7cthndrkgeymhnasunze5pzeaw0khc994sdvn0qrll0r7j7wvxxv3t86ll2x2y8qqqqqqqqqqqqqqqysqqqqqqqqqqqqp6f9jm7k9yqqqqqq2gpr82l0mf2pqkl499zwtaed24kshmqpv4q3m3c24cgwjgujf2c0d9qjh950xf2k25qcpjfvtqggrr9ulum8qu60vumve5j7y57esa5323ju8q0whm5kl4aseupcj0c9lqsxd2npltp2tk26ml2a2dpjhkp6lcllev4ckvfytwhvcsr49xn44zlgl07gzwym3ldccjhlcf69zwmhzafa3avdcmvt3ejkztxg7xcva6
# ->031979fe6ce0e69ece6d99a4bc4a7b30ed22a8cb8703dd7dd2dfaf619e07127e0b->020000000000000000000000000000000000000000000000000000000000000001: Success: preimage=c5897c5e0ff873a228adb7a119fdb949f8c48c83ad46dfbf86ef131e1b54464e
{
   "payment_preimage": "c5897c5e0ff873a228adb7a119fdb949f8c48c83ad46dfbf86ef131e1b54464e",
   "failed_parts": 0,
   "successful_parts": 1,
   "amount_msat": 103000,
   "amount_sent_msat": 103000
}

lightning-cli setconfig xpay-handle-pay false
{
   "config": {
      "config": "xpay-handle-pay",
      "value_bool": false,
      "source": "/media/ssd/.lightning_test/bitcoin/config:2",
      "plugin": "/home/user/lightning_test/plugins/cln-xpay",
      "dynamic": true
   }
}

lightning-cli fetchinvoice lno1zcssxxtelekwpe57eekenf9uffanpmfz4r9cwq7a0hfdltmpncr3ylst 103000
{
   "invoice": "lni1qqgxkqzscynmam6dg0xuf5mt8lfj293pqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqk5srqxf9skppqfjwcqxglc76agrz4x8vr2w5qxj3pltr9myknd0h8p0k0ngdu9s6ngycqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqkqk2d2yyttzr9e55x88vlr4tsled8jegk4z2qftu4dlh72fhyp5ptsqs8xs3qtjn6874vxm7hqfu93ma8lvw9u2wms4khm2e0qkqfgfhtlevqqeqhkq8njtvhjavjjllpwq8ghdxnxydpwaac3plx7epwgfcleasgy0u2pxhx0v62x6sm47mqt2flx4uq6v6y8qqqqqqqqqqqqqqqysqqqqqqqqqqqqp6f9jm7k9yqqqqqq2gpr82l07p2pquf9znuqvkl7r8wnfwctane9guvzlrncynrlrf77960tg3gr6dau65qcpjfvtqggrr9ulum8qu60vumve5j7y57esa5323ju8q0whm5kl4aseupcj0c9lqsxnuwrxgkefd5glee5k33076nm8zm8yjg9qym0ls6qdjysk4f5azvmmlttsdfqpgqmcfpkkufqyw5us0lypr4eszcftjtv378jdwhpcg",
   "changes": {}
}

lightning-cli pay lni1qqgxkqzscynmam6dg0xuf5mt8lfj293pqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqk5srqxf9skppqfjwcqxglc76agrz4x8vr2w5qxj3pltr9myknd0h8p0k0ngdu9s6ngycqvvhnlnvurnfanndnxjtcjnmxrkj92xtsupa6lwjm7hkr8s8zflqkqk2d2yyttzr9e55x88vlr4tsled8jegk4z2qftu4dlh72fhyp5ptsqs8xs3qtjn6874vxm7hqfu93ma8lvw9u2wms4khm2e0qkqfgfhtlevqqeqhkq8njtvhjavjjllpwq8ghdxnxydpwaac3plx7epwgfcleasgy0u2pxhx0v62x6sm47mqt2flx4uq6v6y8qqqqqqqqqqqqqqqysqqqqqqqqqqqqp6f9jm7k9yqqqqqq2gpr82l07p2pquf9znuqvkl7r8wnfwctane9guvzlrncynrlrf77960tg3gr6dau65qcpjfvtqggrr9ulum8qu60vumve5j7y57esa5323ju8q0whm5kl4aseupcj0c9lqsxnuwrxgkefd5glee5k33076nm8zm8yjg9qym0ls6qdjysk4f5azvmmlttsdfqpgqmcfpkkufqyw5us0lypr4eszcftjtv378jdwhpcg
{
   "destination": "031979fe6ce0e69ece6d99a4bc4a7b30ed22a8cb8703dd7dd2dfaf619e07127e0b",
   "payment_hash": "e24a29f00cb7fc33ba697617d9e4a8e305f1cf0498fe34fbc5d3d688a07a6f79",
   "created_at": 1733812199.894748007,
   "parts": 1,
   "amount_msat": 103000,
   "amount_sent_msat": 103000,
   "payment_preimage": "5dc70c964da12e79ca4167f52483364670d8428b0929a9b7ae747c5f292f2998",
   "status": "complete"
}
~                                                    
```

**Comment by rustyrussell** [comment]
This was documented, but not well.  I will fix it, instead!

**Comment by hMsats** [comment]
Solved via #7938

**Comment by rustyrussell** [comment]
Cool, but GH will auto-close when that gets merged, BTW!
